test/bidir-full-sync-tests.js

```diff
@@ -47,6 +48,107 @@ test2('bidirectional full sync test', getClusterSizes(5), 20000,
 	})
 );
 
+test2('bidirectional full sync throttling test', getClusterSizes(3), 20000,
+    prepareCluster(function(t, tc, n) {
+        return [
+            dsl.assertStats(t, tc, n+1, 0, 0),
+            dsl.waitForEmptyPing(t, tc),
+
+            assertReverseFullSyncThrottling(t, tc, 0, n*3, 5),
+
+            dsl.assertStats(t, tc, n+1, 0, 0),
+        ];
+    })
+);
+
+// assert if the number of joins is throttled.
+function assertReverseFullSyncThrottling(t, tc, idx, count, expectedJoins) {
+    var functions = [];
+
+    //pause handling joins to stall reverse full syncs
+    functions.push(pauseHandlingJoins);
+    functions.push(dsl.waitForEmptyPing(t, tc));
+
+    // trigger the reverse full syncs
+    functions.push(triggerReverseFullSyncs(tc, idx, count));
+
+    // verify the number of join requests
+    functions.push(verify);
```

Comment:
Hmm, you're right. Every handler that is registered before this was registered with _registerHandlers. That function adds some magic sauce on the handlers that pushes the every incoming request on the events list.
Can you refactor that so that it uses _registerHandler (singular) and use that function here, in stead of overwriting the handler manually. Then you can use, waitForJoins, and I think if you do that, you don't need to restore the handlers again either.